src/Symfony/Component/Messenger/Exception/ThrownExceptionDetails.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Exception;
+
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+
+/**
+ * ThrownExceptionDetails captures details of a thrown exception and is displayed like a normal exception when dumped
+ * with the VarDumper.
+ *
+ * @author Jeroen Noten <jeroennoten@me.com>
+ */
+class ThrownExceptionDetails
```

Comment:
If this extended `\Exception`, then we would not need to modify the VarDumper component, right?

Reply:
We cannot extend `Exception`, because then the file, line and trace details will not be accurate, we cannot set those details to match the actual thrown exception. Note that this represents an exception that was thrown, stored in the message transport, and then reconstructed to be displayed in the console. This console command has not access to the originally thrown exception.